frontend/app/src/components/defi/Lending.vue

```diff
@@ -117,7 +154,42 @@ export default class Lending extends Vue {
   accountGain!: (address: string) => BigNumber;
   accountGainUsdValue!: (address: string) => BigNumber;
   dsrHistory!: AccountDSRMovement[];
+  dsrHistoryByAddress!: DSRHistory;
   isTaskRunning!: (type: TaskType) => boolean;
+  filteredAccounts: GeneralAccount[] = [];
+
+  async mounted() {
+    await this.$store.dispatch('balances/fetchDSRBalances');
+    if (this.premium) {
+      await this.$store.dispatch('balances/fetchDSRHistory');
+    }
+  }
+
+  get lendingAddresses(): Account[] {
+    let addresses: Account[] = [];
+
+    // join addresses from dsrHistory and dsrBalances
+    const dsrHistoryAddresses = this.dsrHistory.map(dsrEvent => {
+      return { chain: 'ETH', address: dsrEvent.address } as Account;
+    });
+
+    const dsrBalanceAddresses = this.dsrBalances.map(dsrBalance => {
+      return { chain: 'ETH', address: dsrBalance.address } as Account;
```

Comment:
This feels like duplicate work here, wouldn't make more sense to just collect the addresses and then do the mapping to Account?